content/docs/how-we-work/git/branch-merge.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Branching and Merging
+weight: 4
+description: Git branching and merging are essential concepts for managing any software. By using branches, we can isolate different features or bug fixes, and merge them back into the main branch as soon as they are ready.
+---
+
+# Branching and Merging
+
+Git branching and merging are essential concepts for managing any software. By using branches, we can isolate different features or bug fixes, and merge them back into the main branch as soon as they are ready.
+
+## Branching for multiple features
+
+One of the most common uses for git branches is to manage multiple features. When we are working on a large project, it is often helpful to create a separate branch for each feature. This allows us to work on the feature independently without affecting each other. With regard to branch names, we may need to follow a specific nomenclature for branches or be project specific.
+
+To create a branch for a new feature, you can use the following command:
+
+```bash
+git branch my-new-feature
+```
+
+This will create a new branch named `my-new-feature`. You can then switch to the new branch by using the following command:
+
+```bash
+git checkout my-new-feature
+```
+
+Once you are on the new branch, you can start working on implementing the feature. When you are finished, you can push your changes and raise PR.
```

Comment:
@Yemaneberhan-Lemma I see a link for `raise PR` in the document but that is missing here. Also is there a section we can link for `push your changes` text?

The reason am asking is that a lot of engineers are not aware about what is remote and think `origin` is a keyword rather realising it is default remote name GitHub gives. Also they think they have to every time do git push origin branch name where if you are having 1 remote you don't need to mention that in command. Just git push would suffice. These are minor details we can cover in this page. Also we can mention how to manage multiple remotes

Something like https://docs.github.com/en/get-started/using-git/pushing-commits-to-a-remote-repository and https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories

Reply:
I remove the link intentionally because raise PR is linked to PR section which is not in the Eng.HB currently. We will link it after the PR page is done.

Reply:
ok thanks for clarifying. What about the other recommendation for `push your changes` text? If you are having something planned for next iteration then its fine

Reply:
@Kalaiselvan88 We'll also link that to the page that discuss PR details.

Reply:
ok resolving this based on above comments